### PR TITLE
v3.0.0 enhancements

### PR DESCRIPTION
**Pull request by lxghtless**
### Enhancements, Features, Refactoring
- rework code base
- support loading config when used in global module (i.e. cli)
- support loading config from url
- support `get` functionality on config in addition to simply dotting access to property values.
- support manual init mode
	* when auto init is on, config is loaded asynchronously

> The latest changes can be tested with `npm i occams-conf@next`

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -98,3 +127,33 @@ module.exports = {
     resourceUrl: process.env.TEST_URL || 'https://reqres.in/api/mixtapes'
 };
 ```
+
+### Windows Users
+
+If and error resembling the following occurs, please check out the link below it.
```
Comment (misterlantz) [inline readme.md]: `If an error`

**Review by misterlantz — CHANGES_REQUESTED** [review]
Just a grammar mistake in the readme

**Review by seesharpguy — DISMISSED** [review]
after initial npmrc settings issues, I could install and test locally. 👍 Looks good

File: readme.md (hunk 1 above)
Comment (lxghtless) [reply readme.md]: I'm bad at words.

**Review by misterlantz — APPROVED** [review]

**Pull request merged**